bot/exts/utilities/epoch.py

```diff
@@ -0,0 +1,134 @@
+from typing import Optional, Union
+
+import arrow
+import discord
+from dateutil import parser
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.exts.core.extensions import invoke_help_command
+
+# https://discord.com/developers/docs/reference#message-formatting-timestamp-styles
+STYLES = {"Epoch": ("",),
+          "Short Time": ("t", "h:mm A",),
+          "Long Time": ("T", "h:mm:ss A"),
+          "Short Date": ("d", "MM/DD/YYYY"),
+          "Long Date": ("D", "MMMM D, YYYY"),
+          "Short Date/Time": ("f", "MMMM D, YYYY h:mm A"),
+          "Long Date/Time": ("F", "dddd, MMMM D, YYYY h:mm A"),
+          "Relative Time": ("R",)
+          }
+DROPDOWN_TIMEOUT = 60
+
+
+class DateString(commands.Converter):
+    """Convert a relative or absolute date/time string to an arrow.Arrow object."""
+
+    async def convert(self, ctx: commands.Context, argument: str) -> Union[arrow.Arrow, Optional[tuple]]:
+        """
+        Convert a relative or absolute date/time string to an arrow.Arrow object.
+
+        Try to interpret the date string as a relative time. If conversion fails, try to interpret it as an absolute
+        time. Tokens that are not recognised are returned along with the part of the string that was successfully
+        converted to an arrow object. If the date string cannot be parsed, BadArgument is raised.
+        """
+        try:
+            return arrow.utcnow().dehumanize(argument)
+        except ValueError:
+            try:
+                dt, ignored_tokens = parser.parse(argument, fuzzy_with_tokens=True)
+                return arrow.get(dt), ignored_tokens
+            except parser.ParserError:
+                raise commands.BadArgument(f"`{argument}` Could not be parsed to a relative or absolute date")
+
+
+class Epoch(commands.Cog):
+    """Convert an entered time and date to a unix timestamp."""
+
+    @commands.command(name="epoch")
+    async def epoch(self, ctx: commands.Context, *, date_time: DateString = None) -> None:
+        """
+        Convert an entered date/time string to the equivalent epoch.
+
+        **Relative time**
+            accepted units: "seconds", "minutes", "hours", "days", "weeks", "months", "years"
+            Eg ".epoch in a month 4 days and 2 hours"
```

Comment:
A question more than anything, is it an `arrow` limitation that `.epoch 6 hours` and `.epoch in 6 hours` produce different results? In my ideal world, `.epoch 6 hours` would mean the same thing as `.epoch in 6 hours`

Reply:
Yes, it seems a relative time can't be parsed without an `in ..` or `... ago`. I guess it's probably best to mention that in the docstring

Reply:
this could maybe be manually adjusted, by inserting "in" if it isn't there, and ago is not there